propagation.go

```diff
@@ -41,71 +46,124 @@ var (
 type BuiltinFormat byte
 
 const (
-	// SplitBinary encodes the Span in a SplitBinaryCarrier instance.
+	// Binary encodes the Span for propagation as opaque binary data.
+	//
+	// For Tracer.Inject(): the carrier must be an `io.Writer`.
 	//
-	// The `carrier` for injection and extraction must be a
-	// `*SplitBinaryCarrier` instance.
-	SplitBinary BuiltinFormat = iota
+	// For Tracer.Join(): the carrier must be an `io.Reader`.
+	Binary BuiltinFormat = iota
 
-	// SplitText encodes the Span in a SplitTextCarrier instance.
+	// TextMap encodes the Span in a TextMapCarrier instance.
+	//
+	// For Tracer.Inject(): the carrier must be a `TextMapWriter`.
+	//
+	// For Tracer.Join(): the carrier must be a `TextMapReader`.
+	//
+	// See HTTPHeaderTextMapCarrier for an implementation of both TextMapWriter
+	// and TextMapReader that defers to an http.Header instance for storage.
+	// For example, Inject():
+	//
+	//    carrier := HTTPHeaderTextMapCarrier{
+	//    	HeaderPrefix: "opentracing-",
+	//    	Header:       h,
+	//    }
+	//    err := span.Tracer().Inject(span, TextMap, carrier)
+	//
+	// Or Join():
 	//
-	// The `carrier` for injection and extraction must be a `*SplitTextCarrier`
-	// instance.
+	//    carrier := HTTPHeaderTextMapCarrier{
+	//    	HeaderPrefix: "opentracing-",
```

Comment:
this example looks wrong to me, it implies that it's the instrumented library that decides which prefix to add to keys on the wire, while I think it's the tracer who should decide, as otherwise the library of the RPC peer service must use the exact same prefix. I think we only need 
```go
type HTTPHeaderTextMapCarrier http.Header```
```
and add read/write methods that delegate to the Header struct.